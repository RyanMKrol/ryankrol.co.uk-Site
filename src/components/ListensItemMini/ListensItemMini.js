import React, { Component } from 'react';

import styles from './ListensItemMini.module.css';

/**
 * A component representing a subset of information about a "listens" item
 */
class ListensItemMini extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    this.albumLink = props.albumLink;
    this.albumName = props.albumName;
    this.thumbnail = props.thumbnail;

    if (!this.albumLink || !this.albumName || !this.thumbnail) {
      throw new Error('Could not create ListensItemMini');
    }
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX ListensItemMini component
   */
  render() {
    return (
      <a className={styles.listensMiniLink} href={this.albumLink}>
        <img alt={this.albumName} src={this.thumbnail} className={styles.listensItemMiniImage} />
      </a>
    );
  }
}

export default ListensItemMini;
