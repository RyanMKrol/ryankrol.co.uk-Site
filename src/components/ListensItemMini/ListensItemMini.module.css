.listensItemMiniImage,
.listensMiniLink {
  max-width: 100px;
  max-height: 100px;
  transition: all 0.1s ease-in-out;
}

.listensMiniLink:hover {
  transform: scale(1.2);
}
