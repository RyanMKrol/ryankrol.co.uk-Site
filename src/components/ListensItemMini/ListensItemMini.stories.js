import DEFAULT_MOCK from './__mocks__/ListensItemMiniMocks';

export default {
  title: 'Components/ListensItemMini',
};

export const Default = () => DEFAULT_MOCK;
