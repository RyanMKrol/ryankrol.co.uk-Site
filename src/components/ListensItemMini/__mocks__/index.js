import ListensItemMini from '../ListensItemMini';

const DEFAULT_MOCK = (
  <ListensItemMini
    albumLink="https://www.last.fm/music/Kendrick+Lamar/DAMN."
    albumName="DAMN"
    thumbnail="https://lastfm.freetls.fastly.net/i/u/300x300/243a7444e2d59ff9b38e123f7267e3be.jpg"
  />
);

export default DEFAULT_MOCK;
