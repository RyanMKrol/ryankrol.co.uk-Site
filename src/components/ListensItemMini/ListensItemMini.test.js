import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListensItemMini from './ListensItemMini';

import DEFAULT_MOCK from './__mocks__/ListensItemMiniMocks';

it('renders ListensItemMini without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('renders ListensItemMini without throwing', () => {
  expect(() => mount(DEFAULT_MOCK)).not.toThrow();
});

it('ListensItemMini throws without album link', () => {
  expect(() => mount(<ListensItemMini />)).toThrow();
});

it('ListensItemMini throws without album name', () => {
  expect(() => mount(<ListensItemMini albumLink="test-link" />)).toThrow();
});

it('ListensItemMini throws without thumbnail', () => {
  expect(() => mount(<ListensItemMini albumLink="test-link" albumName="test-name" />)).toThrow();
});
