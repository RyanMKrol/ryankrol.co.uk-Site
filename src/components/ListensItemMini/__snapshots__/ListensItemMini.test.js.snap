// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListensItemMini without issue 1`] = `
<a
  className="listensMiniLink"
  href="https://www.last.fm/music/Kendrick+Lamar/DAMN."
>
  <img
    alt="DAMN"
    className="listensItemMiniImage"
    src="https://lastfm.freetls.fastly.net/i/u/300x300/243a7444e2d59ff9b38e123f7267e3be.jpg"
  />
</a>
`;
